includes/admin/class-wcs-admin-meta-boxes.php

```diff
@@ -84,9 +83,13 @@ public function add_meta_boxes() {
 
 		add_meta_box( 'subscription_renewal_orders', __( 'Related Orders', 'woocommerce-subscriptions' ), 'WCS_Meta_Box_Related_Orders::output', 'shop_subscription', 'normal', 'low' );
 
-		// Only display the meta box if an order relates to a subscription
-		if ( 'shop_order' === WC_Data_Store::load( 'order' )->get_order_type( $post_ID ) && wcs_order_contains_subscription( $post_ID, 'any' ) ) {
-			add_meta_box( 'subscription_renewal_orders', __( 'Related Orders', 'woocommerce-subscriptions' ), 'WCS_Meta_Box_Related_Orders::output', 'shop_order', 'normal', 'low' );
+		// Get "Edit Order" screen ID, which differs if HPOS is enabled.
+		$screen         = wcs_is_custom_order_tables_usage_enabled() ? wc_get_page_screen_id( 'shop-order' ) : 'shop_order';
+		$current_screen = get_current_screen();
+
```

Comment:
Added this `$current_screen->id === $screen` condition, as it was rendering on other screens (e.g. Edit Product).